sim/dex-species.ts

```diff
@@ -578,6 +578,7 @@ export class DexSpecies {
 		const movePool = new Set<ID>();
 		for (const {species, learnset} of this.getFullLearnset(id)) {
 			for (const moveid in learnset) {
+				if (this.eggMovesOnly(species, this.get(id))) eggMovesOnly = true;
```

Comment:
```suggestion
				eggMovesOnly = this.eggMovesOnly(species, this.get(id));
```

Reply:
The problem with this change is that `eggMovesOnly` is also assigned to in line 616/617. If that assignment is no longer necessary then `eggMovesOnly` could be inlined into the `if` statement but at the very least its declaration should be moved.

Reply:
I made it so that `eggMovesOnly` on line 616/617 can't be overridden in #10597.